stratagem-runner/src/lib.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+#![allow(non_upper_case_globals)]
+#![allow(non_camel_case_types)]
+#![allow(non_snake_case)]
+#![crate_name = "llapi"]
+
+pub mod raw {
+    include!("bindings.rs");
+}
+use std::convert::From;
+use std::fmt;
+use std::ffi::{CStr, CString};
+use std::io;
+static PATH_BYTES: usize = 4096;
+
+// pub struct Statfs {
+//     pub type: i64;
+//     pub block: i64;
+// }
+pub struct Fid {
+    pub seq: u64,
+    pub oid: u32,
+    pub ver: u32,
+}
+impl fmt::Display for Fid {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "[0x{:x}:0x{:x}:0x{:x}]", self.seq, self.oid, self.ver)
+    }
+}
+// impl From<String> for Fid {
+//     fn from(fidstr: String) -> Self {
+//     }
+// }
+impl From<raw::lu_fid> for Fid {
+    fn from(fid: raw::lu_fid) -> Self {
+        Fid { seq: fid.f_seq,
+              oid: fid.f_oid,
+              ver: fid.f_ver,
+        }
+    }
+}
+
+pub fn fid2path(device: &String, fidstr: &String) -> Option<String> {
+    if !fidstr.starts_with("[") {
```

Comment:
```suggestion
    if !fidstr.starts_with('[') {
```

Reply:
Shouldn't the formatter catch this?

Reply:
@johnsonw rust fmt doesn't but clippy does.